src/command/render/render-shared.ts

```diff
@@ -36,10 +36,17 @@ import {
 } from "../../core/platform.ts";
 import { isProjectInputFile } from "../../project/project-shared.ts";
 
+import { setInitializer, initState } from "../../core/lib/yaml-validation/state.ts";
+import { initPrecompiledModules } from "../../core/lib/yaml-validation/deno-init-precompiled-modules.ts";
+
 export async function render(
   path: string,
   options: RenderOptions,
 ): Promise<RenderResult> {
+  // one time initialization of yaml validators
```

Comment:
The "core" render function that all codepaths end up hitting is `renderFiles()` (for example, `quarto preview` of a project never calls the `render` function but does call `renderFiles()`)

Reply:
Just for completeness - as we discussed, this can't be called on renderFiles because it's too late (project configuration validation has already happened by then). We solved this individually on every quarto command but it needs fixing later.